src/client/auth.js

```diff
@@ -1,16 +1,67 @@
 const UUID = require('uuid-1345')
 const yggdrasil = require('yggdrasil')
+const fs = require('mz/fs')
+const promisify = require('es6-promisify')
+const mkdirp = promisify(require('mkdirp'))
 
-module.exports = function (client, options) {
+module.exports = async function (client, options) {
```

Comment:
If this is a async function maybe we should handle the error ?

Reply:
I'm not sure, I made a nested async function instead to handle the error inside the function, but what is the preferred way to handle errors if not logging to console?